plugins/woocommerce/src/Internal/Orders/OrderDetailsRestController.php

```diff
@@ -0,0 +1,132 @@
+			'/orders/(?P<id>[\d]+)/details',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => fn( $request ) => $this->run( $request, 'send_order_details' ),
+					'permission_callback' => fn( $request ) => $this->check_permissions( $request ),
+					'args'                => $this->get_args_for_send_order_details(),
+					'schema'              => $this->get_schema_for_send_order_details(),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Permission check for REST API endpoint.
+	 *
+	 * @param WP_REST_Request $request The request for which the permission is checked.
+	 * @return bool|WP_Error True if the current user has the capability, otherwise a WP_Error object.
+	 */
+	private function check_permissions( WP_REST_Request $request ) {
+		$order_id = $request->get_param( 'id' );
+		$order    = wc_get_order( $order_id );
+
+		if ( ! $order ) {
+			return new WP_Error( 'woocommerce_rest_not_found', __( 'Order not found', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		return $this->check_permission( $request, 'read_shop_order', $order_id );
+	}
+
+	/**
+	 * Get the accepted arguments for the POST request.
+	 *
+	 * @return array[]
+	 */
+	private function get_args_for_send_order_details(): array {
+		return array(
+			'id' => array(
+				'description' => __( 'Unique identifier of the order.', 'woocommerce' ),
+				'type'        => 'integer',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+		);
+	}
+
+	/**
+	 * Get the schema for both the GET and the POST requests.
+	 *
+	 * @return array[]
+	 */
+	private function get_schema_for_send_order_details(): array {
+		$schema['properties'] = array(
+			'message' => array(
+				'description' => __( 'A message indication whether the email was sent.', 'woocommerce' ),
+				'type'        => 'string',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+		);
+		return $schema;
+	}
+
+	/**
+	 * Handle the POST /orders/{id}/details.
+	 *
+	 * Sends the order details email to the customer.
+	 *
+	 * @param WP_REST_Request $request The received request.
+	 * @return array|WP_Error Request response or an error.
+	 */
+	public function send_order_details( WP_REST_Request $request ) {
+		$order_id = $request->get_param( 'id' );
+		$order    = wc_get_order( $order_id );
+		if ( ! $order ) {
+			return new WP_Error( 'invalid_order', __( 'Invalid order ID.', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		/**
+		 * Fires before an order email is resent.
+		 *
+		 * @since 1.0.0
+		 */
+		do_action( 'woocommerce_before_resend_order_emails', $order, 'customer_invoice' );
+
+		WC()->mailer()->customer_invoice( $order );
+
+		$order->add_order_note( __( 'Order details manually sent to customer.', 'woocommerce' ), false, true );
+
+		/**
+		 * Fires after an order email has been resent.
+		 *
+		 * @since 1.0.0
+		 */
+		do_action( 'woocommerce_after_resend_order_email', $order, 'customer_invoice' );
```

Comment:
I think we could change it to something like "Order details sent to customer via REST API".

Reply:
Makes sense, thanks! Fixed in 8bdffa72b246c6e66003de0f6cb5ab25b8adb065

Reply:
One thing that I think would be worth doing here is adding the missing doc blocks to the actions in their original place in `class-wc-meta-box-order-actions.php`, and then in this new code, just putting a "This action is documented in..." like we do in [other places](https://github.com/woocommerce/woocommerce/blob/29a079e116e36bf39b7e84215a9bfd373c0eab95/plugins/woocommerce/includes/wc-stock-functions.php#L67-L68).

Reply:
Also, in the admin resend function, it looks like we are ensuring that payment gateways and shipping class instances are loaded before sending the customer invoice... do we need to do that in the REST API as well?

Reply:
> One thing that I think would be worth doing here is adding the missing doc blocks to the actions in their original place in class-wc-meta-box-order-actions.php, and then in this new code, just putting a "This action is documented in..." like we do in [other places](https://github.com/woocommerce/woocommerce/blob/29a079e116e36bf39b7e84215a9bfd373c0eab95/plugins/woocommerce/includes/wc-stock-functions.php#L67-L68).

I'll do that 👍 

> Also, in the admin resend function, it looks like we are ensuring that payment gateways and shipping class instances are loaded before sending the customer invoice... do we need to do that in the REST API as well?

True, it seems to be working, fine without it, but it may be needed for some reason I'm missing, should I add it back? 🤔

Reply:
I think I would recommend adding it back, just because there might be some edge case where those classes aren't getting loaded ahead of time and then this could result in a fatal error

Reply:
Done! 85e10f30f78ddac6af7808dda5807d891bb1e289